src/pemsa/cart/pemsa_emitter.cpp

```diff
@@ -97,16 +90,20 @@ function arraylen(t)
  return len 
 end 
 function all(a) 
- if a == nil then 
+ local n = arraylen(a)
+ if a == nil or n == 0 then 
   return function() end 
  end 
  local i = 0
- local n = arraylen(a) 
+ local previous_i = nil
  return function() 
-  i = i + 1 
+  if (a[i] == previous_i) then
```

Comment:
what if `a[0]` is a valid entry in the table? Won't this end up returning that value?

Reply:
You're right, it should start at i = 1